src/Model/ElementDecisionTree.php

```diff
@@ -75,4 +76,89 @@ public function CMSEditFirstStepLink()
             $this->FirstStep()->ID
         );
     }
+
+    /**
+     * To provide all the content block data as JSON to make javascript
+     * based front ends easier to build.
+     *
+     * @return string
+     */
+    public function getBlockJSON()
+    {
+        return json_encode(array_merge(
+            [
+                'blockTitle' => $this->Title,
+                'blockIntro' => ShortcodeParser::get('default')->parse($this->Introduction),
+            ],
+            $this->getTreeData()
+        ));
+    }
+
+    /**
+     * Helper to extract the step and answer data from the block.
+     *
+     * @return array
+     */
+    private function getTreeData()
+    {
+        $data = [
+            'steps' => [],
+            'answers' => [],
+        ];
+
+        // start from the top
+        $first = $this->FirstStep();
+
+        if (!$first->exists()) {
+            return $data;
+        }
+
+        // now start our descent through the flow
+        $this->collectStepData($first, $data);
+
+        // provide way to inject extra data
+        $this->extend('updateTreeData', $data);
+
+        return $data;
+    }
+
+    /**
+     * Helper to collect the data for a given step.
+     *
+     * @param DecisionTreeStep $step
+     * @param array &$data
+     * @return void
+     */
+    private function collectStepData($step, &$data)
+    {
+        // add step data into array
+        $data['steps'][$step->ID] = $step->toJSONData();
+
+        // collect answer data into the array
+        $this->collectAnswersData($step->Answers(), $data);
+    }
+
+    /**
+     * Helper to collect the data for the given answers.
+     *
+     * @param DataList $answers
+     * @param array &$data
+     * @return void
+     */
+    private function collectAnswersData($answers, &$data)
+    {
+        // loop through answers
+        foreach ($answers as $answer) {
+            // push answer data into array
+            $data['answers'][$answer->ID] = $answer->toJSONData();
```

Comment:
Same question here, re not using an index here?

Reply:
as above. and the answer data contains a `goTo`, which is the id of the next step, so you can just reference it directly by key in the `steps` data.